lib/src/features/leaderboard/ui/leaderboard_screen.dart

```diff
@@ -0,0 +1,216 @@
+                        ? SliverGrid(
+                            gridDelegate:
+                                SliverGridDelegateWithFixedCrossAxisCount(
+                                    mainAxisExtent: 644,
+                                    crossAxisCount:
+                                        (constraints.maxWidth / 300).floor()),
+                            delegate:
+                                SliverChildListDelegate(_buildList(context)))
+                        : SliverList(
+                            delegate:
+                                SliverChildListDelegate(_buildList(context)),
+                          )))
+          ]);
+        }));
+  }
+
+  Widget _buildAndroid(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(context.l10n.leaderboard),
+      ),
+      body: LayoutBuilder(
+        builder: (BuildContext context, BoxConstraints constraints) {
+          if (constraints.maxWidth > 600) {
+            return GridView(
+              shrinkWrap: true,
+              gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
+                mainAxisExtent: 644,
+                crossAxisCount: (constraints.maxWidth / 300).floor(),
+              ),
+              children: _buildList(context),
+            );
+          } else {
+            return ListView(
+              children: _buildList(context),
+            );
+          }
+        },
+      ),
+    );
+  }
+
+  List<Widget> _buildList(BuildContext context) {
+    return [
+      _buildLeaderboard(
+          context, leaderboard.bullet, LichessIcons.bullet, 'BULLET'),
+      _buildLeaderboard(
+          context, leaderboard.blitz, LichessIcons.blitz, 'BLITZ'),
+      _buildLeaderboard(
+          context, leaderboard.rapid, LichessIcons.rapid, 'RAPID'),
+      _buildLeaderboard(
+          context, leaderboard.classical, LichessIcons.classical, 'CLASSICAL'),
+      _buildLeaderboard(context, leaderboard.ultrabullet,
+          LichessIcons.ultrabullet, 'ULTRA BULLET'),
+      _buildLeaderboard(
+          context, leaderboard.crazyhouse, LichessIcons.h_square, 'CRAZYHOUSE'),
+      _buildLeaderboard(
+          context, leaderboard.chess960, LichessIcons.die_six, 'CHESS 960'),
+      _buildLeaderboard(context, leaderboard.kingOfThehill, LichessIcons.bullet,
+          'KING OF THE HILL'),
+      _buildLeaderboard(context, leaderboard.threeCheck,
+          LichessIcons.three_check, 'THREE CHECK'),
+      _buildLeaderboard(
+          context, leaderboard.atomic, LichessIcons.atom, 'ATOMIC'),
+      _buildLeaderboard(
+          context, leaderboard.horde, LichessIcons.horde, 'HORDE'),
+      _buildLeaderboard(
+          context, leaderboard.antichess, LichessIcons.antichess, 'ANTICHESS'),
+      _buildLeaderboard(context, leaderboard.racingKings,
+          LichessIcons.racing_kings, 'RACING KINGS'),
+    ];
+  }
+
+  Widget _buildLeaderboard(BuildContext context, List<LightUser> userList,
```

Comment:
Prefer writing a private Stateless widget class instead.
Cf: https://stackoverflow.com/questions/53234825/what-is-the-difference-between-functions-and-classes-to-create-reusable-widgets for more info.
For a one-time private method it's ok to use a function, but for a reusable widget that produces a widget tree, it's better to take the habit to write a class (even if here it is private this widget is reused several times).